Cython/Includes/numpy/__init__.pxd

```diff
@@ -973,22 +974,12 @@ cdef extern from "numpy/ufuncobject.h":
 
     int _import_umath() except -1
 
+cdef inline void set_array_base(object arr, object base):
+    Py_INCREF(base)
+    PyArray_SetBaseObject(arr, base)
 
-cdef inline void set_array_base(ndarray arr, object base):
-     cdef PyObject* baseptr
-     if base is None:
-         baseptr = NULL
-     else:
-         Py_INCREF(base) # important to do this before decref below!
-         baseptr = <PyObject*>base
-     Py_XDECREF(arr.base)
-     arr.base = baseptr
-
-cdef inline object get_array_base(ndarray arr):
-    if arr.base is NULL:
-        return None
-    else:
-        return <object>arr.base
+cdef inline object get_array_base(object arr):
+    return <object>arr.base
```

Comment:
No need to cast here, since this is clearly using Python object access, which returns an … object! :)
But why isn't this using `PyArray_BASE()`? (Probably not expected to occur in performance critical code …)

Reply:
I wanted it to go through the python-level attribute access, maybe that is extreme. I will redo this with PyArray_BASE.